src/net_processing.cpp

```diff
@@ -4104,14 +4137,14 @@ bool PeerManager::SendMessages(CNode* pto)
                     // receiver rejects addr messages larger than MAX_ADDR_TO_SEND
                     if (vAddr.size() >= MAX_ADDR_TO_SEND)
                     {
-                        m_connman.PushMessage(pto, msgMaker.Make(NetMsgType::ADDR, vAddr));
+                        m_connman.PushMessage(pto, msgMaker.Make(msg_type, vAddr));
```

Comment:
If you use the `CNetMsgMaker:: Make(int nFlags, std::string msg_type, Args&&... args)` variant here, you can control the serialization flags, without needing the special casing for ADDRV2 inside `Make`.

Reply:
A good interface is [hard or impossible to misuse](https://testing.googleblog.com/2018/07/code-health-make-interfaces-hard-to.html). If we add `ADDRV2_FORMAT` to `nFlags` inside `Make()` that makes it impossible for the caller to misuse it like:

```cpp
msgMaker.Make(0, NetMsgType::ADDRv2, vAddr); // huh!?
```

Reply:
@vasild I agree with the notion, but this is just the wrong layer to deal with that. The serialization framework (which I'd consider netmessagemaker to be a thin wrapper around) shouldn't have knowledge about the data being serialized, or even about which message types exist.

> I think it is not good to impose on the caller: "whenever you pass `NetMsgType::ADDRv2` for `msg_type` you must also pass `ADDRV2_FORMAT` in `nFlags`".

I think that's the wrong way to think about it. `CNetMsgMaker` should do as it's told: serializing the object passed with the specified flags, and nothing else. Knowledge about what message types exist, and how data is serialized in them belongs in net_processing.

Reply:
That is also a valid point. Changed.